### ISSUE TITLE
Hardcoded limit for Roche Potential

### ISSUE THREAD

**Issue by saultyevil** [sirocco-rt/sirocco#475]
For binary models, there are a fair number of numerical routines carried out to construct a pillbox around the secondary star. A lot of the routines are either finding some maxima or roots of the Roche potential for the binary system. As the routines, found in roche.c, were originally designed for WDs there are a few hardcoded maximum limits set in some of the routines.

Notably, when calculating the actual Roche potential with the `phi` function (line 456 roche.c), there is a floor which sets the Roche potential to be no smaller than -1e20. In a standard CV model, this is not a big problem. But, in the TDE models where the mass of the primary is ~1e7 Msun, we run into the problem where the calculated potential far exceeds the hardcoded limit. Culminating in a bunch of errors within some of our numerical routines, i.e. rtsafe. To stop any undefined behaviour with the numerical routines, I've added in a call to exit to stop it from producing bogus results. 

As an enhancement for our treatment of binary systems, we should probably be setting this hardcoded limit more intelligently with some sort of calculated characteristic Roche potential. This could possibly be done by setting a limit such as G * reduced mass / smallest characteristic length scale.

**Comment by kslong** [comment]
@saultyevil   This does not sound hard to handle, but can you either fix this, or provide a .pf file that tests whether we have achieved success.

**Comment by kslong** [comment]
@saultyevil   This is the sort of thing we should not leave hanging around.  Can I renew my request that you either fix this (preferred since you have understood this already), or put a .pf file that demonstrates the problem.

**Comment by saultyevil** [comment]
Sure, I had already started working on this but became distracted (sorry!). For completeness sake, I'll also attach a pf which demonstrates this problem.

[tde_binary.pf.txt](https://github.com/agnwinds/python/files/3151818/tde_binary.pf.txt)

**Comment by saultyevil** [comment]
I think this problem may be a little more complicated than I first thought.

Changing the limit for the Roche potential so it doesn't truncate *does* fix one problem during the binary setup routines. But, the binary routines now fall over elsewhere. For example, I have tried *not* truncating the value of the Roche potential but this now means rtsafe doesn't converge towards a value for the Lagrange L1 point, when previously it did.

I had also tried increasing the x range for rtsafe when calculating the L2 point just in case the L2 point wasn't in the given range of [1.01 * geo.a, 2.0 * geo.a]. This sometimes worked, but was sensitive to changes to the truncation value of the Roche potential.

I eventually found some reasonable choices for the truncation value and the L2 point x range. Now rtsafe was able to converge towards a value for L1 and L2. But it failed in roche2_width_max. I think it was failing here because when the value of the function at either end of the bracketed interval [x1, x2] was the same, i.e. f(x1) = f(x2) = Roche potential limit.

So I think I need to be a bit more clever...

**Comment by saultyevil** [comment]
The original problem was that our root finding algorithm was failing. Since Nick's hard work replacing all of our numerical routines for issue #542 with those in GSL, the error I described in the first post where Python would crash doesn't happen anymore. I assume that the GSL routines are doing a better job at handling the root finding.

To confirm this, I looked closer at the routines and the return codes of GSL. Basically, the GSL routines are now returning GSL_SUCCESS for the same troublesome parameter files where previously a root could not be found. So it seems that the crashing issue has been solved at least.

I note that I haven't done what was originally intended - which was to set intelligent limits on the Roche potential rather than a hardcoded value. But, since the code is not crashing anymore, maybe we can finally close this issue? I'll leave it open for now to reflect on it a little more.
